### PR TITLE
BIGTOP-4098: Drop python2 in bigtop-toolchain

### PR DESCRIPTION
**Pull request by JiaLiangC**
### Description of PR
Drop python2 in bigtop-toolchain and bigtop-deploy

The logic here includes creating a default Python. I believe the logic for creating the default Python should be handled within bigtop-deploy, so it has been removed.

### How was this patch tested?
manual test
image

### For code changes:

- [ ] Does the title or this PR starts with the corresponding JIRA issue id (e.g. 'BIGTOP-3638. Your PR title ...')?
- [ ] Make sure that newly added files do not have any licensing issues. When in doubt refer to https://www.apache.org/licenses/

### CODE REVIEW

**Comment by JiaLiangC** [comment]
@iwasakims Could you help review this pr?

**Comment by iwasakims** [comment]
bigtop-deploy seems not to be the target here. How about update the title of PR and JIRA?

**Comment by JiaLiangC** [comment]
@iwasakims pr title and issue already updated

File: bigtop_toolchain/manifests/packages.pp [hunk 1]
```diff
@@ -293,21 +293,9 @@
...
-      if (($operatingsystem == 'Ubuntu' and 0 <= versioncmp($operatingsystemmajrelease, '22.04'))) {
-        file { '/usr/bin/python':
-          ensure => 'link',
-          target => '/usr/bin/python2',
-        }
-        $pkgs = $_pkgs
-      } elsif (($operatingsystem == 'Ubuntu' and 0 <= versioncmp($operatingsystemmajrelease, '20.04')) or ($operatingsystem == 'Debian' and 0 <= versioncmp($operatingsystemmajrelease, '11'))) {
-        $pkgs = concat($_pkgs, ["python-is-python2"])
-      } else {
-        $pkgs = $_pkgs
```
Comment (iwasakims) [inline bigtop_toolchain/manifests/packages.pp:309]: This breaks build on Ubuntu and Debian due to undefined `pkgs`. @JiaLiangC.

```
Warning: Unknown variable: 'pkgs'. (file: /home/admin/srcs/bigtop/bigtop_toolchain/manifests/packages.pp, line: 305, column: 13)
Error: Evaluation Error: Missing title. The title expression resulted in undef (file: /home/admin/srcs/bigtop/bigtop_toolchain/manifests/packages.pp, line: 305, column: 13) on node ip-172-31-163-160.ap-northeast-1.compute.internal

> Task :toolchain FAILED

FAILURE: Build failed with an exception.
```

Comment (JiaLiangC) [reply bigtop_toolchain/manifests/packages.pp:309]: @iwasakims  fixed
image

File: bigtop_toolchain/manifests/packages.pp [hunk 2]
```diff
@@ -293,21 +293,10 @@
...
-      }
+      $pkgs = $_pkgs
```
Comment (iwasakims) [inline bigtop_toolchain/manifests/packages.pp:299]: We should remove no longer needed temporary `_pkgs` now. @JiaLiangC.

Comment (JiaLiangC) [reply bigtop_toolchain/manifests/packages.pp]: @iwasakims  already remove redundant variable $_pkgs

**Review by iwasakims — APPROVED** [review]
LGTM. Ran toolchain task on centos-7-x86_64, fedora-38-aarch64, debian-11-x86_64 and ubuntu-22.04-x86-64. Thanks, @JiaLiangC.

**Comment by uscxiexing** [comment]
image
When I use openeuler 22.03 to compile alluxio, the webUI module reports an error that node-gyp requires python2.But openeuler without python2. The compilation fails.

**Pull request merged**